### PR TITLE
AzureMonitor: Apply query migrations in QueryEditor

### PR DESCRIPTION
**Pull request by joshhunt**
**What this PR does / why we need it**:

Ports query migrations from the old angular controller to react.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

NOTE, this isn't finished yet - the migrateToDefaultNamespace is triggering when you create a new query, so I want to understand it's intention and sort that out.

### CODE REVIEW

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/components/QueryEditor/usePreparedQuery.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+/**
+ * Returns queries with some defaults + migrations, and calls onChange function to notify if it changes
+ */
+const usePreparedQuery = (query: AzureMonitorQuery, onChangeQuery: (newQuery: AzureMonitorQuery) => void) => {
+  const preparedQuery = useMemo(() => prepareQuery(query), [query]);
+
+  useEffect(() => {
+    if (preparedQuery !== query) {
+      console.log('changing default query', { from: query, to: preparedQuery });
```
Comment (joshhunt) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/components/QueryEditor/usePreparedQuery.ts:33]: remove

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/types/query.ts [hunk 2]
```diff
@@ -51,6 +54,11 @@ export interface AzureMetricQuery {
 
   /** @deprecated Remove this once angular is removed */
   allowedTimeGrainsMs?: number[];
+
+  /** @deprecated migrated to dimensionFilters */
+  dimension?: string;
+  /** @deprecated migrated to dimensionFilters */
```
Comment (sunker) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/types/query.ts:60]: remove this comment?

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/types/query.ts [hunk 3]
```diff
@@ -51,6 +54,11 @@ export interface AzureMetricQuery {
...
+  dimensionFilter?: string;
```
Comment (sunker) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/types/query.ts:61]: Why migrating from dimension to dimensionFilter? Is it to align it better with the api?

**Review by sunker — COMMENTED** [review]
Looks good to me. 

Should we also migrate the annotation query?

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/types/query.ts (hunk 3 above)
Comment (joshhunt) [reply public/app/plugins/datasource/grafana-azure-monitor-datasource/types/query.ts:61]: I'm not sure!

This are all old and existing migrations that the Angular query_ctrl.js has been doing for ages, and needs to be moved so they can happen in the react QueryEditor now that Angular is not in the picture any more.

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/types/query.ts (hunk 2 above)
Comment (joshhunt) [reply public/app/plugins/datasource/grafana-azure-monitor-datasource/types/query.ts:60]: I want the types to remain so I don't have to cast to any to access them in the migration, but I want the `@deprecated` annotation and comment so people know not to try and access it elsewhere.

**Review by sunker — APPROVED** [review]
Not tested, but looks good!

**Pull request merged**